packages/client/src/containers/ShoppingCartPage/ShoppingCartPage.container.js

```diff
@@ -51,7 +74,11 @@ export const ShoppingCartPage = () => {
         </div>
         <CartCount productSum={productSum} />
         <CartContainer>
-          <CartTable products={products} />
+          <CartTable
+            products={products}
+            setProducts={setProducts}
```

Comment:
Why is this table able to set the products directly? If the table component is used to add/remove/change quantity/filter then it should expose those events instead. It should not have access to the state that ShoppingCartPage owns.

Reply:
I'm not sure if I understood correctly, but previous mentors insisted that not all states be placed in the parent component.